### PR TITLE
En_Sellnuts

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c [hunk 1]
```diff
@@ -22,99 +98,1072 @@ const ActorInit En_Sellnuts_InitVars = {
...
-static InitChainEntry D_80ADDB00[] = {
+static ActorAnimationEntryS D_80ADD990[] = {
+    { &D_06005488, 1.0f, 0, -1, 0, 0 },  { &D_0600B0B4, 1.0f, 0, -1, 0, 0 },  { &D_06004AA0, 1.0f, 0, -1, 2, 0 },
+    { &D_06004E38, 1.0f, 0, -1, 2, 0 },  { &D_060029E8, 1.0f, 0, -1, 2, 0 },  { &D_06005CA8, 1.0f, 0, -1, 2, 0 },
+    { &D_060038CC, 1.0f, 0, -1, 0, 0 },  { &D_06003CC0, 1.0f, 0, -1, 0, 0 },  { &D_060012F4, 1.0f, 0, -1, 2, 0 },
+    { &D_06004700, 1.0f, 0, -1, 2, 0 },  { &D_06001BC8, 1.0f, 0, -1, 0, 0 },  { &D_06003438, 1.0f, 0, -1, 2, 0 },
+    { &D_06001E2C, 1.0f, 0, -1, 0, 0 },  { &D_06000994, 1.0f, 0, -1, 0, 0 },  { &D_06002268, 1.0f, 0, -1, 2, 0 },
+    { &D_06002F08, 1.0f, 0, -1, 0, 0 },  { &D_0600577C, 1.0f, 0, -1, 0, -4 }, { &D_060029E8, 1.0f, 8, -1, 2, 0 },
+    { &D_060029E8, 1.0f, 4, -1, 2, -4 }, { &D_060029E8, 1.0f, 0, -1, 2, 0 },  { &D_06001BC8, 1.0f, 0, -1, 0, 0 },
+    { &D_060012F4, -1.0f, 0, -1, 2, 0 }, { &D_06002670, 1.0f, 0, -1, 2, 0 },
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c:128]: This is an amazing number of animations for this actor, I guess as a haggler he gesticulates more than I realized.

**Review by hensldm — APPROVED** [review]
LGTM

File: src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c [hunk 2]
```diff
@@ -22,99 +98,1072 @@ const ActorInit En_Sellnuts_InitVars = {
...
+    }
+}
+
+void func_80ADB254(EnSellnuts* this, GlobalContext* globalCtx) {
+    s16 currentFrame = this->skelAnime.animCurrentFrame;
+    s16 frameCount = SkelAnime_GetFrameCount(&D_80ADD990[this->unk_34C].animationSeg->common);
+
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 2, 0xE38);
+    if ((((this->actor.yDistToPlayer < 50.0f) && (this->actor.yDistToPlayer > -50.0f)) ? 1 : 0) &&
+        ((this->actor.xzDistToPlayer < 200.0f) ? 1 : 0)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c:318]:
```suggestion
    if (((this->actor.yDistToPlayer < 50.0f) && (this->actor.yDistToPlayer > -50.0f) &&
        (this->actor.xzDistToPlayer < 200.0f)) {
```
Are the `? 1 : 0` really needed to match? I've never seen that before. If they are needed, then I think `? true : false` would be fine

File: src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c [hunk 3]
```diff
@@ -22,99 +98,1072 @@ const ActorInit En_Sellnuts_InitVars = {
...
+                    this->actionFunc = func_80ADBAB8;
+                    break;
+            }
+        }
+    } else {
+        goto label;
+    label: // fake
+        if (((this->actor.xzDistToPlayer < 80.0f) &&
+             (((this->actor.yDistToPlayer < 50.0f) && (this->actor.yDistToPlayer > -50.0f)) ? 1 : 0)) ||
+            (this->actor.isTargeted)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c:409]:
```suggestion
             (((this->actor.yDistToPlayer < 50.0f) && (this->actor.yDistToPlayer > -50.0f)) ? 1 : 0)) ||
            this->actor.isTargeted) {
```
Same comment as befoore for the `? 1 : 0`

File: src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c [hunk 4]
```diff
@@ -22,99 +98,1072 @@ const ActorInit En_Sellnuts_InitVars = {
...
+            } else if (gSaveContext.weekEventReg[17] & 0x40) {
+                this->unk_340 = D_80ADD918[this->unk_33A];
+            } else {
+                this->unk_340 = D_80ADD910[this->unk_33A];
+            }
+        }
+    }
+
+    if (!(((this->actor.yDistToPlayer < 50.0f) && (this->actor.yDistToPlayer > -50.0f)) ? 1 : 0) ||
+        !((this->actor.xzDistToPlayer < 200.0f) ? 1 : 0)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c:436]: Same as above

File: src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c [hunk 5]
```diff
@@ -22,99 +98,1072 @@ const ActorInit En_Sellnuts_InitVars = {
...
+            return;
+        }
+    } else if ((this->unk_366 == 1) && (this->unk_368 == 20)) {
+        ActorCutscene_Stop(this->cutscene);
+        this->cutscene = ActorCutscene_GetAdditionalCutscene(this->cutscene);
+        ActorCutscene_SetIntentToPlay(this->cutscene);
+        this->unk_366 = 2;
+    }
+
+    this->unk_368 += 1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c:715]:
```suggestion
    this->unk_368++;
```

File: src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c [hunk 6]
```diff
@@ -22,99 +98,1072 @@ const ActorInit En_Sellnuts_InitVars = {
...
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sellnuts/func_80ADC580.s")
+void EnSellnuts_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnSellnuts* this = THIS;
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sellnuts/func_80ADC5A4.s")
+    this->unk_328 += 1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c:1061]:
```suggestion
    this->unk_328++;
```

File: src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c [hunk 7]
```diff
@@ -22,99 +98,1072 @@ const ActorInit En_Sellnuts_InitVars = {
...
+void func_80ADD7CC(GlobalContext* globalCtx, s32 limbIndex, Actor* thisx) {
+    EnSellnuts* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sellnuts/EnSellnuts_Update.s")
+    if (((this->unk_350 == 1) || (this->unk_350 == 3)) && ((limbIndex == 23) || (limbIndex == 24))) {
+        Matrix_Scale(this->unk_354, this->unk_358, this->unk_35C, MTXMODE_APPLY);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sellnuts/func_80ADD4E0.s")
+    if (((this->unk_34C == 9) || (this->unk_34C == 0x14) || (this->unk_34C == 22)) &&
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c:1153]:
```suggestion
    if (((this->unk_34C == 9) || (this->unk_34C == 20) || (this->unk_34C == 22)) &&
```
For consistency

**Review by engineer124 — COMMENTED** [review]
Good work! Only a few small comments

File: src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c [hunk 8]
```diff
@@ -22,99 +98,1072 @@ const ActorInit En_Sellnuts_InitVars = {
...
+                    break;
+
+                default:
+                    this->actionFunc = func_80ADBAB8;
+                    break;
+            }
+        }
+    } else {
+        goto label;
+    label: // fake
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c:406]: Would this match without the `goto`?
```suggestion
    label: // fake
```

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**